mpt/core/src/main/java/org/apache/james/mpt/protocol/ProtocolSession.java

```diff
@@ -497,6 +502,24 @@ public boolean isClient() {
         }
     }
 
+    private class ReinitElement implements ProtocolElement {
+
+        private final int sessionNumber;
+
+        public ReinitElement(int sessionNumber) {
+            this.sessionNumber = sessionNumber < 0 ? 0 : sessionNumber;
+        }
+
+        public void testProtocol(Session[] sessions, boolean continueAfterFailure) throws Exception {
+            Session session = sessions[sessionNumber];
```

Comment:
I didn't try to think about the current behaviour, but negative is expected in some cases (sessionNumber=-1 is explicitely passed) while session.length is expected to be checked outside